Feat/5 labelling process

joelgrand opened the pull request:
* Added functionallity
 - to label a datapoint
 - to remove the labeled datapoint
 - API tests for testing behaviour
* Added new Migrations with DB changes
* Small enhancement in other tests

Note: I already merged the old version of Gianbellos branch into mine.

guentgia commented on D-lama-service/Data/ProjectEntities/DataPoint.cs:
```diff
@@ -10,12 +10,23 @@ public abstract class DataPoint : Entity
+        // Required foreign key property
+        public int ProjectId { get; set; }
+        // Required reference navigation to principal
+        public Project Project { get; set; } = null!;
```
I would add the Annotation [JsonIgnore] for the Project reference i order to avoid circular references during the json serialization.
```suggestion
        [JsonIgnore]
        public Project Project { get; set; } = null!;
```

joelgrand replied:
Good point. Added your suggestion

guentgia commented on D-lama-service/Data/ProjectEntities/Project.cs:
```diff
@@ -11,23 +11,30 @@ public class Project : Entity
         public User Owner { get; set; } = null!;
+
         // Collections navigation containing dependents
         public ICollection<TextDataPoint> TextDataPoints { get; } = new List<TextDataPoint>();
```
As we now only have a table for the entity DataPoint (instead of two separate ones for TextDataPoint and ImageDataPoint), we can change this List to DataPoint.
```suggestion
        public ICollection<DataPoint> DataPoints { get; } = new List<DataPoint>();
```

guentgia commented on D-lama-service/Data/ProjectEntities/DataPoint.cs:
```diff
@@ -10,12 +10,23 @@ public abstract class DataPoint : Entity
         public int Version { get; set; } // increment per update
+
+        // Required foreign key property
+        public int ProjectId { get; set; }
```
Good idea to make the FK refernce in DataPoint already. Could you please then delete the navigation in TextDataPoint? It is not needed there now (remove lines 14-19 in file TextDataPoint.cs)

joelgrand commented:
> Thank you very much for this great feature implementation Gioello! I added some comments with suggestions just for completing the changes to the Entities. All my manual tests worked (also with the suggested modifications) :)

Thanks for the feedback Gianbello. I have added the missing things! Please have a look again.

guentgia reviewed (commented):
Thanks for the great job!
